app/lib/features/pins/pages/create_pin_page.dart

```diff
@@ -0,0 +1,353 @@
+import 'package:acter/common/providers/space_providers.dart';
+import 'package:acter/common/themes/app_theme.dart';
+import 'package:acter/common/utils/routes.dart';
+import 'package:acter/common/utils/utils.dart';
+import 'package:acter/common/widgets/html_editor.dart';
+import 'package:acter/common/widgets/input_text_field.dart';
+import 'package:acter/common/widgets/md_editor_with_preview.dart';
+import 'package:acter/common/widgets/sliver_scaffold.dart';
+import 'package:acter/common/widgets/spaces/select_space_form_field.dart';
+import 'package:acter/features/home/providers/client_providers.dart';
+import 'package:acter/features/pins/pin_utils/pin_utils.dart';
+import 'package:acter/features/pins/providers/pins_provider.dart';
+import 'package:acter/features/settings/providers/settings_providers.dart';
+import 'package:acter_flutter_sdk/acter_flutter_sdk_ffi.dart';
+import 'package:appflowy_editor/appflowy_editor.dart';
+import 'package:atlas_icons/atlas_icons.dart';
+import 'package:flutter/material.dart';
+import 'package:flutter_easyloading/flutter_easyloading.dart';
+import 'package:flutter_riverpod/flutter_riverpod.dart';
+import 'package:go_router/go_router.dart';
+
+class CreatePinPage extends ConsumerStatefulWidget {
+  final String? initialSelectedSpace;
+  static const titleFieldKey = Key('create-pin-title-field');
+  static const contentFieldKey = Key('create-pin-content-field');
+  static const urlFieldKey = Key('create-pin-url-field');
+  static const submitBtn = Key('create-pin-submit');
+
+  const CreatePinPage({super.key, this.initialSelectedSpace});
+
+  @override
+  ConsumerState<CreatePinPage> createState() => _CreatePinSheetConsumerState();
+}
+
+class _CreatePinSheetConsumerState extends ConsumerState<CreatePinPage> {
+  final TextEditingController _titleController = TextEditingController();
+  final TextEditingController _linkController = TextEditingController();
+  final TextEditingController _textController = TextEditingController();
```

Comment:
```suggestion
  final TextEditingController _textController = TextEditingController();
  final TextEditingController _htmlTextController = TextEditingController();
```

See below for context.